### PR TITLE
Many updates to example/

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
FYI @sethladd – here's the rendered readme: https://github.com/dart-lang/json_serializable/tree/more_readme_fun/example/

Thoughts?

File: example/README.md [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
+* See [`lib/example.dart`][example] for an example of a file using these
+  annotations.
+
+* See [`lib/example.g.dart`][example_g] for the generated file.
+
+Run the build script to update the generated code after you modify of of the
+source files.
+
+```console
+> dart bin/build.dart
```
Comment (kevmoo) [inline example/README.md]: @sethladd – how does a flutter dev invoke a dart script?

File: example/test/readme_test.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+void main() {
+  _expect('README.md');
+  _expect('pubspec.yaml');
+}
+
+void _expect(String fileName) {
+  test(fileName, () {
+    final file = new File(fileName);
+    expect(file.readAsStringSync(), contains(_pubspecContent));
```
Comment (natebosch) [inline example/test/readme_test.dart]: what is the value here? Making sure that versions are kept up to date? Should you read the pubspecs instead and use the versions from there?

Comment (kevmoo) [reply example/test/readme_test.dart]: Then I have to read that file and figure out which lines I want. This seemed easier.

File: example/README.md [hunk 3]
```diff
@@ -1,23 +1,57 @@
...
+[INFO] BuildDefinition: Reading cached asset graph completed, took 54ms
+[INFO] BuildDefinition: Checking for updates since last build completed, took 663ms
+[INFO] Build: Running build completed, took 10ms
+[INFO] Build: Caching finalized dependency graph completed, took 44ms
+[INFO] Build: Succeeded after 68ms with 0 outputs
+```
+
+Assuming you follow the layout of the [`tool`][tool] directory, run
```
Comment (jakemac53) [inline example/README.md]: the rest of the sentence appears to be missing here :)

File: example/README.md (hunk 1 above)
Comment (sethladd) [reply example/README.md]: today, it's a bit gnarly (@mit-mit is looking to improve this), but it's:

`flutter packages pub run package:script`

See https://github.com/flutter/flutter/issues/13360 which asks to make this better.

Comment (kevmoo) [reply example/README.md]: Sadly, this won't work. It's not in the bin directory, which is the only place `pub run` will run scripts.

File: example/README.md (hunk 3 above)
Comment (kevmoo) [reply example/README.md]: ack

**Comment by kevmoo** [comment]
Everyone: PTAL

File: example/README.md [hunk 4]
```diff
@@ -1,23 +1,68 @@
...
+```console
+> dart bin/build.dart
+[INFO] BuildDefinition: Reading cached asset graph completed, took 54ms
+[INFO] BuildDefinition: Checking for updates since last build completed, took 663ms
+[INFO] Build: Running build completed, took 10ms
+[INFO] Build: Caching finalized dependency graph completed, took 44ms
+[INFO] Build: Succeeded after 68ms with 0 outputs
+```
+
+*NOTE*: If you're using Flutter, running a Dart script is a bit tricky. You can
```
Comment (kevmoo) [inline example/README.md]: @sethladd @mit-mit – here's the best I could come up with.

Comment (kevmoo) [reply example/README.md]: Follow-up bug: https://github.com/flutter/flutter/issues/13607

**Pull request merged**